src/Stratis.Bitcoin/Configuration/NodeSettings.cs

```diff
@@ -41,13 +40,27 @@ public class NodeSettings
 
         /// <summary>
         /// Initializes a new instance of the object.
-        /// <para>This constructor does not load the configuration itself.</para>
         /// </summary>
-        public NodeSettings()
+        /// <param name="name">Blockchain name. Currently only "bitcoin" and "stratis" are used.</param>
+        /// <param name="innerNetwork">Specification of the network the node runs on - regtest/testnet/mainnet.</param>
+        /// <param name="protocolVersion">Supported protocol version for which to create the configuration.</param>
+        /// <param name="agent">The nodes user agent that will be shared with peers.</param>
+        public NodeSettings(string name = "bitcoin", Network innerNetwork = null, ProtocolVersion protocolVersion = SupportedProtocolVersion, string agent = "StratisBitcoin")
```

Comment:
I suggest we move Name, and ProtocolVersion in to Network, those are constants per blockchain, agent can indeed change.